src/state.py

```diff
@@ -307,15 +314,19 @@ def from_charm(  # pylint: disable=too-many-locals
             error_field_str = " ".join(f"{f}" for f in error_fields)
             raise RecoverableStateError(f"Invalid charm configuration {error_field_str}") from exc
         custom_config_ssh_key = _fetch_ssh_repository_key(charm.model, valid_config)
-        agent_password = _fetch_agent_password(charm.model, valid_config)
+        agent_password = _fetch_password(charm.model, valid_config.agent_password)
+        api_password = _fetch_password(charm.model, valid_config.api_password)
```

Comment:
Since the password can't be changed, we should be able to see if the value changed and at least inform the user that they can't change its value and that it's the previous value which is applied

Reply:
Where is the error raised if the password was already changed ?

Reply:
"The password change should always happen and will occur only once,", otherwise the API will return an error, that will result in a WazuhInstallationError

Reply:
Let's close this thread in favour of the other one that open